x/programs/rust/examples/nft/src/metadata.rs

```diff
@@ -0,0 +1,35 @@
+//! NFT schema
+//! See https://nftschool.dev/reference/metadata-schemas/#ethereum-and-evm-compatible-chains for more information
+//! on the ERC-721 NFT metadata schema.
+
+use serde_derive::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Debug, Default)]
+pub struct Nft {
+    pub symbol: String,
+    pub name: String,
+    pub uri: String,
+    pub edition: i64,
+}
+
+impl Nft {
+    pub fn with_symbol(mut self, symbol: String) -> Self {
+        self.symbol = symbol;
+        self
+    }
+
+    pub fn with_name(mut self, name: String) -> Self {
+        self.name = name;
+        self
+    }
+
+    pub fn with_uri(mut self, uri: String) -> Self {
+        self.uri = uri;
+        self
+    }
+
+    pub fn with_edition(mut self, edition: i64) -> Self {
+        self.edition = edition;
+        self
+    }
+}
```

Comment:
I think the builder pattern here is unnecessary for an example. It's easier to read the four-property struct on its own.

Reply:
I expect the number of fields to grow